### PR TITLE
Port meson from deprecated python3

### PR DESCRIPTION
**Pull request by infirit**
@ChenQi1989 try this and then set  `-Dpython.purelibdir`

### CODE REVIEW

**Comment by infirit** [comment]
Fixed bunch of other stuff like prefix.

File: meson.build [hunk 1]
```diff
@@ -25,8 +25,8 @@ datadir = join_paths([prefix, get_option('datadir')])
 pkgdatadir = join_paths([prefix, get_option('datadir'), package_name])
 bindir = join_paths([prefix, get_option('bindir')])
 libexecdir = join_paths([prefix, get_option('libexecdir')])
-schemadir = join_paths([datadir, 'glib-2.0', 'schemas'])
```
Comment (eli-schwartz) [inline meson.build:28]: You don't need to care about making schemadir absolute. It is only used as the install_dir to a function. Meson functions for installing always accept relative paths (e.g. `share/glib-2.0/schemas`) and install the files relative to the prefix.

You do need to ensure this is an absolute path if you intend to do something else with schemadir, such as configure it into a file.

Comment (infirit) [reply meson.build:28]: Make sense, fixed. In this case I made a big mess as datadir already has prefix :fearful: 

At some point I'll rework the directory handling and pass relative paths and absolute when configuring.

**Comment by ChenQi1989** [comment]
@infirit I've tested these patches in my cross compilation environment. They worked well. Thanks.

File: meson.build [hunk 2]
```diff
@@ -47,10 +47,11 @@ conf_data.set('icondir', join_paths(datadir, 'icons'))
 conf_data.set('dhconfig', get_option('dhcp-config-path'))
 conf_data.set('POLKIT', have_polkit)
 conf_data.set('GETTEXT_PACKAGE', package_name)
-conf_data.set('PYTHON', pyprog.path())
+conf_data.set('PYTHON', pyinstall.path())
 
 # Check for build dependencies
-pythonlib = dependency('python-' + python.language_version(), required: true)
+#pythonlib = dependency('python-' + pyinstall.language_version(), required: true)
```
Comment (cschramm) [inline meson.build:53]: Drop?

**Review by cschramm — APPROVED** [review]

File: meson.build (hunk 2 above)
Comment (infirit) [reply meson.build]: Yes, forgot to remove. Fixed

**Pull request merged**